feat(oxfmt): Support more extensions from linguist-languages

leaysgur opened the pull request:
Part of #10573

Prettier supports many more file extensions than those listed in `oxc_span::VALID_EXTENSIONS`.

Refs:

- https://github.com/ikatyang-collab/linguist-languages/blob/main/data/JavaScript.js
- https://github.com/ikatyang-collab/linguist-languages/blob/main/data/TypeScript.js

At the same time, I simplified the file extension check to a simple function.

Boshen commented:
### Merge activity

* **Sep 9, 8:58 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

leaysgur committed:
feat(oxfmt): Support more extensions from linguist-languages (#13607)

Part of #10573

Prettier supports many more file extensions than those listed in `oxc_span::VALID_EXTENSIONS`.

Refs:

- https://github.com/ikatyang-collab/linguist-languages/blob/main/data/JavaScript.js
- https://github.com/ikatyang-collab/linguist-languages/blob/main/data/TypeScript.js

At the same time, I simplified the file extension check to a simple function.
